source/Calamari.Common/Features/StructuredVariables/StructuredConfigVariablesService.cs

```diff
@@ -49,6 +49,19 @@ public class StructuredConfigVariablesService : IStructuredConfigVariablesServic
             this.log = log;
 
             allReplacers = replacers;
+            
+#if NET40
+            // Temporary fix for dependency resolution happening backwards for .NET 4.0 versus 4.5.2 and dotnet
```

Comment:
It might be worth adding in a test (potentially in a follow up PR) that checks the correct behaviour for .net 4.0, to protect against any regressions?

Reply:
Agree - will get on it as soon as I can once the fix is rolled out.